src/components/infinite-scroll/infinite-scroll.ts

```diff
@@ -267,7 +286,23 @@ export class InfiniteScroll {
    * to `enabled`.
    */
   complete() {
-    if (this.state === STATE_LOADING) {
+    if (this._position === POSITION_TOP) {
```

Comment:
can you explain a little bit the algorithm used here? 
how many frames does it take? and why?

I think I get it, but still... you probably know better